action/action-util/src/main/java/com/powsybl/action/util/GeneratorUtil.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.action.util;
+
+import com.powsybl.iidm.network.Bus;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Terminal;
+
+/**
+ * @author Olivier Perrin <olivier.perrin at rte-france.com>
+ */
+final class GeneratorUtil {
+
+    private GeneratorUtil() {
+    }
+
+    static void connectGenerator(Generator g) {
+        Terminal t = g.getTerminal();
+        t.connect();
+        if (g.isVoltageRegulatorOn()) {
+            Bus bus = t.getBusView().getBus();
+            if (bus != null) {
+                // set voltage setpoint to the same as other generators connected to the bus
+                double targetV = bus.getGeneratorStream().findFirst().map(Generator::getTargetV).orElse(Double.NaN);
+                // if no other generator connected to the bus, set voltage setpoint to network voltage
+                if (Double.isNaN(targetV) && !Double.isNaN(bus.getV())) {
+                    g.setTargetV(bus.getV());
```

Comment:
I know it's not yours, but it looks like the method does not do what the comment says : 
it does not set the target to the same as other generators target when it finds one.

In the case where the voltage is lower than the target voltage of connected generators, the action may not do what's expected.

Also we can wonder if it is desirable in that case to ignore the targetV that was possibly defined in the generator modification.

Reply:
Maybe we should add a specific unit test for this, once we have decided how it should behave.

Reply:
> I know it's not yours, but it looks like the method does not do what the comment says :
> it does not set the target to the same as other generators target when it finds one.

We could rename  it "setTargetVoltageFromState()" ?

Reply:
> In the case where the voltage is lower than the target voltage of connected generators, the action may not do what's expected.

You talk about calculated voltage?

Reply:
> Also we can wonder if it is desirable in that case to ignore the targetV that was possibly defined in the generator modification.

In ``GeneratorModificationTask.modify(Network, ComputerManager)``, "targetV" is modified after the generator connection if a target V value is specified. Thus, it won't be ignored.

I will add a test for this specific case.

Reply:
> 
> 
> > In the case where the voltage is lower than the target voltage of connected generators, the action may not do what's expected.
> 
> You talk about calculated voltage?

Yes : if calculated voltage is lower than target because generators are at their maximum reactive power, it would seem mor logical to re-use the target instead of the computed value, which may be too low (even lower than operational limits).

Reply:
>>>In the case where the voltage is lower than the target voltage of connected generators, the action may not do what's expected.
>>>
>>You talk about calculated voltage?
>>
>Yes : if calculated voltage is lower than target because generators are at their maximum reactive power, it would seem mor logical to re-use the target instead of the computed value, which may be too low (even lower than operational limits).

I think there is a missing instruction here. If others generators are found, no targetV is set to the current generator, which is pretty weird.

Maybe adding something like
```java
if (!Double.isNaN(targetV)) {
   g.setTargetV(targetV);
}
```
before the other test would solve the problem: we use the targetV of other generators, if any, or the computed voltage if none.

This would be compliant with the comments in the method.

Reply:
Agreed!